src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/messaging/PrisonerReleasedFromPrisonEventService.kt

```diff
@@ -43,36 +46,36 @@ class PrisonerReleasedFromPrisonEventService(
 
   private fun PrisonerReleasedAdditionalInformation.processPrisonerReleaseEvent() {
     log.info { "Processing Prisoner Released From Prison Event for prisoner [$nomsNumber]" }
-    try {
-      reviewScheduleService.exemptActiveReviewScheduleStatusDueToPrisonerRelease(
-        prisonNumber = nomsNumber,
-        prisonId = prisonId,
-      )
-    } catch (e: ReviewScheduleNotFoundException) {
-      log.debug { "Prisoner [$nomsNumber] does not have an active Review Schedule; no need to set it as exempt" }
-    }
-
-    // TODO - RR-1215 - call inductionScheduleService to exempt Induction Schedule due to prisoner release
+    handle(
+      REVIEW_SCHEDULE,
+    ) { reviewScheduleService.exemptActiveReviewScheduleStatusDueToPrisonerRelease(nomsNumber, prisonId) }
+    handle(
+      INDUCTION_SCHEDULE,
+    ) { inductionScheduleService.exemptActiveInductionScheduleStatusDueToPrisonerRelease(nomsNumber, prisonId) }
   }
 
   private fun PrisonerReleasedAdditionalInformation.processPrisonerReleaseEventDueToDeath() {
     log.info { "Processing Prisoner Released From Prison Event (due to death) for prisoner [$nomsNumber]" }
-    try {
-      reviewScheduleService.exemptActiveReviewScheduleStatusDueToPrisonerDeath(
-        prisonNumber = nomsNumber,
-        prisonId = prisonId,
-      )
-    } catch (e: ReviewScheduleNotFoundException) {
-      log.debug { "Prisoner [$nomsNumber] does not have an active Review Schedule; no need to set it as exempt" }
-    }
+    handle(
+      REVIEW_SCHEDULE,
+    ) { reviewScheduleService.exemptActiveReviewScheduleStatusDueToPrisonerDeath(nomsNumber, prisonId) }
+    handle(
+      INDUCTION_SCHEDULE,
+    ) { inductionScheduleService.exemptActiveInductionScheduleStatusDueToPrisonerDeath(nomsNumber, prisonId) }
+  }
 
+  private fun handle(
```

Comment:
This is a little bit fancy - basically instead of repeating a similar try-catch-log 4 times this only does it once by passing in the function and a constant that describes the thing it is.

Reply:
Nice, I like that a lot! 😁 👍